File: HashMap-Dictionary/CountFrequency.java
import java.util.*;

class CountFrequency{
    public static Map<String, Integer> countFreq(String arr[]){
        Map<String, Integer> frequencyMap = new HashMap<>();
        for(String str : arr){
            if(frequencyMap.containsKey(str)){
                frequencyMap.put(str, frequencyMap.get(str) + 1);
            } else {
                frequencyMap.put(str, 1);
            }
                
            }
            return frequencyMap;            
        }

        public static void main(String args[]){
            Scanner sc = new Scanner(System.in);
            String[] input = new String[5];
            System.out.println("Enter array elements");
            for(int i=0; i<5; i++){
                input[i] = sc.nextLine();
            }
            Map<String, Integer> result = countFreq(input);
            System.out.println(result);

            sc.close();
        }
        
    }

File: HashMap-Dictionary/CountFrequency.py
class CountFrequency:
    def count_freq(self, arr):
        frequency = {}
    
        for strings in arr:
            if strings in frequency:
                frequency[strings] += 1
            else:
                frequency[strings] = 1
    
        return frequency
    
if __name__ == "__main__":
    input = ["apple", "orange", "banana", "banana", "apple", "grapes"]
    freq = CountFrequency()
    print(f"Frequency counts: {freq.count_freq(input)}")

    

